lib/Runtime/Library/BuiltIn/BuiltIn.js

```diff
@@ -0,0 +1,60 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function (intrinsic) {
+    var platform = intrinsic.BuiltIn;
+
+    let FunctionsEnum = {
+        ArrayIndexOf: { className: "Array", methodName: "indexOf", argumentsCount: 1 },
+    };
+
+    platform.registerFunction(FunctionsEnum.ArrayIndexOf, function (searchElement, fromIndex) {
+        let o = __chakraLibrary.Object(this);
```

Comment:
Yes it's the object constructor. Actually from what I understand here, [https://tc39.github.io/ecma262/#sec-toobject](https://tc39.github.io/ecma262/#sec-toobject) , we throw an error in case of Null or Undefined value, then we simply instantiate a new Object. I was think that our object constructor was already doing that, but it's not checking for null or undefined value. I fixed the problem.

---
In reply to: [115388233](https://github.com/Microsoft/ChakraCore/pull/2898#discussion_r115388233) [](ancestors = 115388233)